docs/install/kustomize-structure.md

```diff
@@ -0,0 +1,37 @@
+# About the kustomization structure
+
+The kustomization directories are slightly more complex than the average operator due to the dependancy on `cass-operator`. We deploy `cass-operator` using the manifests from Github to ensure that we remain up to date.
+
+## Objectives
+
+We want to achieve the following:
+
+1. Minimal modifications to the `config/` directory initially created by `operator-sdk`.
```

Comment:
We aren't saying we can't, we're saying we should keep changes minimal. This makes upgrading the operator-sdk and other components easier IMO and increases the benefits of familiarity with the framework. 

Additionally, kustomize allows changes to be made in a second location e.g. `aux-config` with exactly the same functionality.

Reply:
I disagree. We don't upgrade something from operator-sdk (kubebuilder) just by "recreating a basic struct". We should update what we need. We create automated RBACs from our annotations, but our overlays (such as cluster-scoped installation) are highly customized just for our use-case.

Reply:
I also disagree with the object. See https://github.com/k8ssandra/cass-operator/issues/167 for one example why. 

Adding additional kustomization directories and overlays in the `config` directory does make upgrading operator-sdk any more complicated.

Reply:
> I disagree. We don't upgrade something from operator-sdk (kubebuilder) just by "recreating a basic struct". We should update what we need. We create automated RBACs from our annotations, but our overlays (such as cluster-scoped installation) are highly customized just for our use-case.

I don't understand the comment `We don't upgrade something from operator-sdk (kubebuilder) just by "recreating a basic struct".` You'd have to unpack this for me.

There is an upgrade [guide](https://sdk.operatorframework.io/docs/upgrading-sdk-version/version-upgrade-guide/) which we should be following. Making modifications minimal will make following it less risky. That's the intention here.

> I also disagree with the object. See k8ssandra/cass-operator#167 for one example why.

Yes, these are the types of issues I'm trying to overcome here. I don't disagree with your point, and that's why I've said **minimal** modifications, not **no modifications**. I don’t see that we are disagreeing here.

It would be good to assemble some set of known modifications we want to make to operator-sdk kustomize scaffolding. Removing namespaces is a great starting point. I am happy to document these as a part of this README so that we know where areas of high change reside.

> Adding additional kustomization directories and overlays in the config directory does make upgrading operator-sdk any more complicated.

It potentially does. Moreover it causes other issues, like I've just experienced where cass-operator and k8ssandra-operator were not playing nice because each had a `leader-election-role` defined. 

While the root cause of that issue is the presence of two bases with conflicting resources at the same level of the kustomization, it is far easier to understand the separation of the base and overlays if both are clearly delineated by their own directory structure.

Reply:
I mean, there's an alternative here too where it is just a free-for-all and we accept all modifications. I guess if that's the case, I'd want to see all of the unused manifests removed so that the `config/` folder is as simple as possible. 

E.g. we don't use the auth proxy but all the manifests still appear in reaper-operator, cass-operator, k8ssandra-operator. The samples directories are often incomplete (that might need to be fixed anyway) scorecard isn't in use AFAIK, in some other operator repos we aren't using the webhook and should be removing that.

I guess the question is - do we just accept that there is cruft in the framework and accept it because it is standardised cruft, or do we want to commit to going over this with a fine toothed comb? 

([Cross linking](https://github.com/k8ssandra/reaper-operator/issues/67) to reaper-operator where I originally did propose this latter approach)

Reply:
Just saw @jsanda's comment. Forum is a good idea. Deleting my above two comments to repost there.

> @Miles-Garnsey @burmanm please move the discussion to https://forum.k8ssandra.io/t/establishing-conventions-with-kustomize/338. I think it has gone beyond the original scope of the PR and is better suited for the forum.